eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,160 @@
+parameters:
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoOwner
+    type: string
+  - name: SdkRepoName
+    type: string
+
+  - name: SdkRepoNameForSpecGen
+    type: string
+  - name: SdkRepoBranch
+    type: string
+    default: 'main'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'AutoRest'
+  - name: TspConfigPath
+    type: string
+  - name: ReadmePath
+    type: string
+  - name: ApiVersion
+    type: string
+    default: 'not-specified'
+  - name: CodePushRequired
+    type: boolean
+  - name: SdkPrRequired
+    type: boolean
+  - name: ApiViewRequired
+    type: boolean
+
+jobs:
+- job:
+  displayName: 'SDK Generation'
+  variables:
+    - template: /eng/pipelines/templates/variables/image.yml
```

Comment:
We might need to add this template to the spec repo as only eng/common folder is synced to spec repo.